### ISSUE TITLE
parameter sets: GROUPS/HEADING

### ISSUE THREAD

**Issue by tehn** [monome/norns#491]
currently we can make SEPARATOR fields for grouping parameters within sets.

it'd be nice to have a HEADING to describe the group.

perhaps some nice UI where the HEADING stays sticky at the top of the screen when scrolling through?

and also not being able to land on a HEADING or SEPARATOR as an active field?

**Comment by tehn** [comment]
issue generated re: clarifying AUX/INSERT for audio params.

**Comment by tehn** [comment]
alternative feature: GROUPS for parameters @markwheeler

**Comment by tehn** [comment]
additional UI ideas @markwheeler 

<img width="792" alt="params-idea" src="https://user-images.githubusercontent.com/246896/45840543-322ad280-bce5-11e8-926f-e95ca4a51352.png">

**Comment by markwheeler** [comment]
Here's our current working proposal:

**New: Param Sections**
- Non-selectable headings in the param list define 'sections' of params.
- Navigate between section headings by holding KEY1 and using ENC2.
- While holding KEY1, the current section heading is displayed at the top of the screen.

**New: Param Groups**
- Sub-pages for the params menu.
- Only one level deep.
- Groups can contain separators
- Adding params into groups will look something like:
```
params:group("sample")
params:add{file: "file" ... }
params:add{control: "start"}
params:group_close()
params:add{control: "whatever"}
params:group("filter")
...
```

**Update: Mapping**
- An item at the top of the param list is used to enable/disable mapping mode.
- In mapping mode, CC mapping works as it currently does.
- In mapping mode, params are displayed using their IDs instead of names for ease of mapping OSC.

**Update: Small things**
- Cycle through options on KEY3.
- Separators are still available and work as they currently do except are non-selectable.

**Open Questions**
- Can sub-pages/groups contain sections?
- Does every param have to be in a section? Might need to be if using for nav.
- Should we name groups something else to avoid confusion with sections?

![params-03](https://user-images.githubusercontent.com/192943/46319997-24046e00-c591-11e8-8ab4-df32abe05314.png)

**Comment by tehn** [comment]
> Should we name groups something else to avoid confusion with sections?

agreed... any ideas? subgroup?

> Can sub-pages/groups contain sections?

this seems unnecessary. imagine holding KEY1 while inside a subgroup... you'd see three layers:

SECTION / SUBGROUP / SUBSECTION

is that overkill? i think it might be.

> Does every param have to be in a section? Might need to be if using for nav.

i think yes. by default a paramset can initialize the current section to "script" (or perhaps the name of the script?) calling `params:section("new section") would change the section name (this first section name could be overwritten, is what i mean).

it makes sense that the top section is called "system" (i think?) and perhaps the default view begins at "script" (so you'd scroll up to the other stuff? or is that very bad design... perhaps).

---

do we have a design idea for the subpage of SAVE / LOAD? (i have an idea... can make a drawing... considering unifying SAVE/LOAD onto the same screen)

MAP is a simple OPTION toggle (on/off)

---

one more proposition: quick keys for SAVE / LOAD (let's call it PSET or FILE?) and MAP:

- when KEY1 is being held, we see the SECTION/SUBPAGE listing up top (inside an overlaid box?)
- below this, two boxes, left/right (trying to visually associate with KEY2 / KEY3) can be PSET (left) and MAP (right).
- PSET jumps to the PSET screen. leaving the screen returns to current scroll position
- MAP shows map mode on/off, and this quick key toggles it

basically these two functions are very interruptive if you have to scroll back to the top of the list, even if you have the SECTION jump. but it's nice to have them also in the menu so there are less mysterious key combinations.

**Comment by markwheeler** [comment]
Cool I agree on all the above.
I would favor starting the screen at the top though (not auto-scrolling to script section).

For save/load design - take a look at the screen I just made for Loom and see if some of that might be applicable. Could potentially also integrate file naming using the keyboard.

And yes I think we can make a nice overlay with prompts for save/load and map.

**Comment by markwheeler** [comment]
Maybe just `group` and `sub_group` or `section` and `sub_section` ?

**Comment by tehn** [comment]
oh this makes a lot of sense!

barely leaning towards section and subsection, would go with a group decision.
